input/postgres/establish_connection.go

```diff
@@ -42,36 +43,49 @@ func EstablishConnection(ctx context.Context, server *state.Server, logger *util
 
 func connectToDb(ctx context.Context, config config.ServerConfig, logger *util.Logger, globalCollectionOpts state.CollectionOpts, databaseName string) (*sql.DB, error) {
 	var dbPasswordOverride string
+	var hostOverride string
+	var sslmodeOverride string
+	var db *sql.DB
+	driverName := "postgres"
 
 	if config.DbUseIamAuth {
-		if config.SystemType != "amazon_rds" {
-			return nil, fmt.Errorf("IAM auth is only supported for Amazon RDS and Aurora - turn off IAM auth setting to use password-based authentication")
-		}
-		sess, err := awsutil.GetAwsSession(config)
-		if err != nil {
-			return nil, err
-		}
-		if dbToken, err := rdsutils.BuildAuthToken(
-			fmt.Sprintf("%s:%d", config.GetDbHost(), config.GetDbPortOrDefault()),
-			config.AwsRegion,
-			config.GetDbUsername(),
-			sess.Config.Credentials,
-		); err != nil {
-			return nil, err
+		if config.SystemType == "amazon_rds" {
+			sess, err := awsutil.GetAwsSession(config)
+			if err != nil {
+				return nil, err
+			}
+			if dbToken, err := rdsutils.BuildAuthToken(
+				fmt.Sprintf("%s:%d", config.GetDbHost(), config.GetDbPortOrDefault()),
+				config.AwsRegion,
+				config.GetDbUsername(),
+				sess.Config.Credentials,
+			); err != nil {
+				return nil, err
+			} else {
+				dbPasswordOverride = dbToken
+			}
+		} else if config.SystemType == "google_cloudsql" {
+			if config.GcpProjectID == "" || config.GcpRegion == "" || config.GcpCloudSQLInstanceID == "" {
+				return nil, fmt.Errorf("To use IAM auth with Google Cloud SQL, you must specify project ID, region, and instance ID")
+			}
+			hostOverride = strings.Join([]string{config.GcpProjectID, config.GcpRegion, config.GcpCloudSQLInstanceID}, ":")
+			// When using cloud-sql-go-connector, this needs to be set as disable
```

Comment:
This is a good write-up for why this needs to be disabled.
I can also add the link as a comment too if that helps:
https://github.com/GoogleCloudPlatform/cloud-sql-go-connector/issues/889